components/match2/commons/starcraft_starcraft2/match_group_input_starcraft.lua

```diff
@@ -89,55 +131,103 @@ function StarcraftMatchGroupInput._readDate(matchArgs)
 end
 
 ---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	local maps = {}
+	local subGroup = 0
+	for mapKey, rawMapInput, mapIndex in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local mapInput = Json.parseIfString(rawMapInput)
+		local map
+		map, subGroup = MapFunctions.readMap(mapInput, subGroup, #opponents)
+
+		map.participants = MapFunctions.getParticipants(mapInput, opponents)
+
+		map.mode = MapFunctions.getMode(mapInput, map.participants, opponents)
+
+		Table.mergeInto(map.extradata, MapFunctions.getAdditionalExtraData(map, map.participants))
+
+		map.vod = Logic.emptyOr(mapInput.vod, match['vodgame' .. mapIndex])
+
+		table.insert(maps, map)
+		match[mapKey] = nil
+	end
+
+	return maps
+end
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInput.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param opponent table
 ---@return table
-function StarcraftMatchGroupInput._checkFinished(match)
-	if Logic.readBoolOrNil(match.finished) == false then
-		match.finished = false
-		return match
-	elseif Logic.readBool(match.finished) then
-		match.finished = true
-	elseif Logic.isNotEmpty(match.winner) then
-		match.finished = true
+function MatchFunctions.getOpponentExtradata(opponent)
+	return {
+		advantage = tonumber(opponent.advantage),
+		penalty = tonumber(opponent.penalty),
+		score2 = opponent.score2,
+		isarchon = opponent.isarchon,
+	}
+end
+
+---@param player table
+---@return string
+function MatchFunctions.getPlayerFaction(player)
+	return player.extradata.faction or Faction.defaultFaction
+end
+
+---@param props {walkover: string|integer?, winner: string|integer?, score: string|integer?, opponentIndex: integer}
+---@param opponent table
+---@param autoScore? fun(opponentIndex: integer): integer?
+function MatchFunctions.computeOpponentScore(props, opponent, autoScore)
+	-- TODO: bot the usage away
+	if props.score == '-' then
+		props.score = 'L'
 	end
 
-	-- Match is automatically marked finished upon page edit after a
-	-- certain amount of time (depending on whether the date is exact)
-	if match.finished ~= true then
-		local threshold = match.dateexact and 30800 or 86400
-		if match.timestamp + threshold < NOW then
-			match.finished = true
-		end
+	local calculatedScore
+	calculatedScore, opponent.status = MatchGroupInput.computeOpponentScore(props, autoScore)
+
+	if Logic.isNumeric(calculatedScore) and not opponent.score then
+		calculatedScore = calculatedScore + (opponent.extradata.advantage or 0) - (opponent.extradata.penalty or 0)
 	end
 
-	return match
+	opponent.score = calculatedScore
+
+	if opponent.status == MatchGroupInput.STATUS.SCORE and (props.winner == 'draw' or tonumber(props.winner) == 0) then
+		opponent.status = MatchGroupInput.STATUS.DRAW
+	end
```

Comment:
Why is this needed? Doesn't commmons take care of this?

Reply:
no, commons doesn't account for it

Reply:
Seems like reasonable thing to add to `isDraw` then

Reply:
(you cna make it in a sep. PR)